e2e/testcases/workload_identity_test.go

```diff
@@ -267,105 +344,236 @@ func TestWorkloadIdentity(t *testing.T) {
+						nsRef: nsChart.Name})); err != nil {
+					nt.T.Fatal(err)
 				}
-			} else {
-				err := nt.WatchForAllSyncs(nomostest.WithRootSha1Func(tc.rootCommitFn),
-					nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{nomostest.DefaultRootRepoNamespacedName: syncDir}))
-				if err != nil {
+				if err = nt.Validate(truncateStringByLength(fmt.Sprintf("%s-%s", rootChart.Name, rootChart.Name), 63),
+					"default", &appsv1.Deployment{},
+					testpredicates.IsManagedBy(nt.Scheme, declared.RootReconciler, rsRef.Name)); err != nil {
+					nt.T.Fatal(err)
+				}
+				if err = nt.Validate(truncateStringByLength(nsChart.Name, 63),
+					nsRef.Namespace, &appsv1.Deployment{},
+					testpredicates.IsManagedBy(nt.Scheme, declared.Scope(nsRef.Namespace), nsRef.Name)); err != nil {
 					nt.T.Fatal(err)
 				}
-				kustomizecomponents.ValidateAllTenants(nt, string(declared.RootReconciler), "../base", tenant)
 			}
 
 			// Migrate from gcpserviceaccount to k8sserviceaccount
 			if tc.testKSAMigration {
-				if err := migrateFromGSAtoKSA(nt, rs, ksaRef, tc.fleetWITest, tc.rootCommitFn); err != nil {
+				if err := migrateFromGSAtoKSA(nt, tc.fleetWITest, tc.sourceType, rsRef, nsRef, tc.newRootSrcCfg, tc.newNSSrcCfg); err != nil {
 					nt.T.Fatal(err)
 				}
 			}
 		})
 	}
 }
 
+type rootSyncMutator func(rs *v1beta1.RootSync)
+type repoSyncMutator func(rs *v1beta1.RepoSync)
+
+func rootSyncUsingHelm(nt *nomostest.NT, rsRef types.NamespacedName, sc sourceConfig, mutators ...rootSyncMutator) (*registryproviders.HelmPackage, error) {
+	chart, err := nt.BuildAndPushHelmPackage(rsRef,
+		registryproviders.HelmSourceChart(sc.chart),
+		registryproviders.HelmChartVersion(sc.version))
+	if err != nil {
+		return nil, fmt.Errorf("pushing helm chart: %w", err)
+	}
+	nt.T.Log("Update RootSync to sync from a helm chart")
+	rootSyncHelm := nt.RootSyncObjectHelm(configsync.RootSyncName, chart)
+	rootSyncHelm.Spec.Helm.ReleaseName = chart.Name
+	for _, mutator := range mutators {
+		mutator(rootSyncHelm)
+	}
+	if err = nt.KubeClient.Apply(rootSyncHelm); err != nil {
+		return nil, err
+	}
+	return chart, nil
+}
+
+func repoSyncUsingHelm(nt *nomostest.NT, rsRef types.NamespacedName, sc sourceConfig, mutators ...repoSyncMutator) (*registryproviders.HelmPackage, error) {
+	chart, err := nt.BuildAndPushHelmPackage(rsRef,
+		registryproviders.HelmSourceChart(sc.chart),
+		registryproviders.HelmChartVersion(sc.version))
+	if err != nil {
+		return nil, fmt.Errorf("pushing helm chart: %w", err)
+	}
+	nt.T.Log("Update RepoSync to sync from a helm chart")
+	repoSyncHelm := nt.RepoSyncObjectHelm(rsRef, chart)
+	repoSyncHelm.Spec.Helm.ReleaseName = chart.Name
+	for _, mutator := range mutators {
+		mutator(repoSyncHelm)
+	}
+	if err = nt.KubeClient.Apply(repoSyncHelm); err != nil {
+		return nil, err
+	}
+	return chart, nil
+}
+
+func rootSyncUsingOCI(nt *nomostest.NT, rsRef types.NamespacedName, sc sourceConfig, mutators ...rootSyncMutator) (rsyncValidateMeta, error) {
```

Comment:
nit: I think the function names here (`r*SyncUsing*`) are a bit imprecise, particularly since they are scoped to the `testcases` package. That being said, I'm not really sure what to name these given that they are making several assumptions that seem pretty specific to this test case

I think something like `SetupRootSyncUsingOCIDelegatedControl` might more accurately describe the behavior, although even then the inputs/outputs still seem fairly bespoke to this test

Reply:
SGTM